vsintegration/src/FSharp.Editor/Common/LanguageService.fs

```diff
@@ -376,38 +373,11 @@ and
             | Some (hier, _) ->
                 match hier with
                 | :? IProvideProjectSite as siteProvider when not (IsScript(filename)) -> 
-                    this.SetupProjectFile(siteProvider, workspace)
+                    this.SetupProjectFile(siteProvider, this.Workspace)
                 | _ -> 
                     let fileContents = VsTextLines.GetFileContents(textLines, textViewAdapter)
-                    this.SetupStandAloneFile(filename, fileContents, workspace, hier)
+                    this.SetupStandAloneFile(filename, fileContents, this.Workspace, hier)
             | _ -> ()
         | _ -> ()
 
-        // This is the second half of the bridge between the F# IncrementalBuild analysis engine and the Roslyn analysis
-        // engine.  When a document gets the focus, we call FSharpLanguageService.Checker.StartBackgroundCompile for the
-        // project containing that file. This ensures the F# IncrementalBuild engine starts analyzing the project.
-        let wpfTextView = textViewAdapter.GetWpfTextView(textView)
-        match wpfTextView.TextBuffer.Properties.TryGetProperty<ITextDocument>(typeof<ITextDocument>) with
-        | true, textDocument ->
-            let filePath = textDocument.FilePath
-            let onGotFocus = new EventHandler(fun _ _ ->
-                for documentId in workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath) do 
-                    let document = workspace.CurrentSolution.GetDocument(documentId)
-                    match projectInfoManager.TryGetOptionsForEditingDocumentOrProject(document) with 
-                    | Some options -> checkerProvider.Checker.StartBackgroundCompile(options)
-                    | None -> ()
-            )
-            let rec onViewClosed = new EventHandler(fun _ _ -> 
-                wpfTextView.GotAggregateFocus.RemoveHandler(onGotFocus)
-                wpfTextView.Closed.RemoveHandler(onViewClosed)
-            )
-            wpfTextView.GotAggregateFocus.AddHandler(onGotFocus)
-            wpfTextView.Closed.AddHandler(onViewClosed)
-        | _ -> ()
-
-        // When the text view is opened, setup all remaining projects on the background.
-        async {
-            do this.BatchSetupProjects()
-        }
-        |> Async.StartImmediate
```

Comment:
I tried Async.StartImmediate in the handler but it doesn't speedup things at all.